### PR TITLE
Us#558 & Us#572

### PR DESCRIPTION
**Pull request by snehi23**
**Taiga User Stories**
US#558 : As a clinician, I want a confirmation when I create a patient 
US#572 : As a clinician, I want to know what filter I am using while viewing tables

**Taiga Task(s)**
#605 Create popup screen on new patient create event
#606 Display PIN on popup screen

#607 Fix filter drop down list

**What did you change?**
US#558
1. Created new dialog view named 'remember-patient-pin'
2. Capture new patient enrolment event
3. pop-up dialog to show necessary details on enrolment

US#572
1. Fixed drop down list problem with jQuery

**How can we test?**
US#558
1. Enrol new patient and after enrolment dialog will appear as per attached shot
2. No dialog for already created patient 
screen shot 2016-04-08 at 4 32 17 pm

US#572
1. Trial to use filter on dashboard view.
2. Selected value will appear on selection in filter
screen shot 2016-04-08 at 4 40 30 pm

### CODE REVIEW

File: view/partial/remember-patient-pin.hbs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+                      Close
+                  </span>
+              </button>
+              <h4 class="modal-title">Patient {{ patient.pin }} Created!</h4>
+            </div>
+            <div class="modal-body">
+              <p>Please copy the pin number {{ patient.pin }} down so information can be re-identified later.</p>
+            </div>
+            <div class="modal-footer">
+              <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
```
Comment (ChristianMurphy) [inline view/partial/remember-patient-pin.hbs]: move text on line 13, 16 and 19, onto a separate newline from tags. (e.g. how lines 9-11 are)

File: view/partial/remember-patient-pin.hbs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                      &times;
+                  </span>
+                  <span class="sr-only">
+                      Close
+                  </span>
+              </button>
+              <h4 class="modal-title">Patient {{ patient.pin }} Created!</h4>
+            </div>
+            <div class="modal-body">
+              <p>Please copy the pin number {{ patient.pin }} down so information can be re-identified later.</p>
```
Comment (poojaRal) [inline view/partial/remember-patient-pin.hbs]: Keep the message 
"Please copy down the PIN {{ patient.pin}}. This information is very important."

File: view/partial/remember-patient-pin.hbs (hunk 1 above)
Comment (snehi23) [inline view/partial/remember-patient-pin.hbs]: ok.

**Comment by ChristianMurphy** [comment]
Also there is a merge conflict that needs to be resolved.

File: view/partial/remember-patient-pin.hbs (hunk 2 above)
Comment (snehi23) [inline view/partial/remember-patient-pin.hbs]: ok will do it.
@ChristianMurphy  are you fine with new message as you only suggested current committed one ?

Comment (ChristianMurphy) [inline view/partial/remember-patient-pin.hbs]: yep :+1:

File: view/partial/remember-patient-pin.hbs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                      Close
+                  </span>
+              </button>
+              <h4 class="modal-title">
+                  Patient {{ patient.pin }} Created!
+              </h4>
+            </div>
+            <div class="modal-body">
+              <p>
+                  Please copy down the PIN {{ patient.pin}}. This information is very important.
```
Comment (ChristianMurphy) [inline view/partial/remember-patient-pin.hbs]: nit, have a space before and after double curlies. e.g. `{{ patient.pin }}`.
Also there is a merge conflict that needs to be resolved.

Comment (snehi23) [inline view/partial/remember-patient-pin.hbs]: ok. will do it.

**Comment by ChristianMurphy** [comment]
LGTM

**Comment by ChristianMurphy** [comment]
CI is throwing an error
```
  ✖ handler-test › create-patient › when patient is created   
  t.is(path, '/patient/1001')
       |                     
       "/patient/1001?newPatient=true"
```

could you update this test with the new redirect url.
https://github.com/promisasu/Portal/blob/development/controller/handler-test/create-patient.js#L86-L91

File: view/script/patient.js [hunk 4]
```diff
@@ -1,5 +1,9 @@
 (function patient () {
     'use strict';
+
+    var isNewPatient = window.location.search;
+    var re = /newPatient=true/;
```
Comment (poojaRal) [inline view/script/patient.js]: Instead of var name as "re", can this be anything relevant?

File: controller/handler/create-patient.js [hunk 5]
```diff
@@ -94,7 +94,7 @@ function createPatient (request, reply) {
         return transaction.commit();
     })
     .then(() => {
-        return reply.redirect(`/patient/${newPatient.pin}`);
+        reply.redirect(`/patient/${newPatient.pin}?newPatient=true`);
```
Comment (ChristianMurphy) [inline controller/handler/create-patient.js]: return the `reply` here

File: view/script/patient.js (hunk 4 above)
Comment (snehi23) [inline view/script/patient.js]: @poojaRal Its Regex. We can name it whatever we want it to be. What do you suggest ?

File: view/script/dashboard.js [hunk 6]
```diff
@@ -4,6 +4,14 @@
     var nameColumn = 0;
     var table = $('#trials').DataTable({order: [nameColumn, 'desc']});
     var strategy = filterStrategyFactory();
+    var selectedChoice = null;
+
+    $('.dropdown-menu').on('click', 'div', function selectItem () {
+        selectedChoice = $(this).text().trim();
```
Comment (ChristianMurphy) [inline view/script/dashboard.js]: remove `var selectedChoice = null` on line 7.
and add `var` in front of line 10.

File: view/script/patient.js (hunk 4 above)
Comment (ChristianMurphy) [inline view/script/patient.js]: maybe `isNewPatientRegex`?

Comment (snehi23) [inline view/script/patient.js]: sure.

File: controller/handler/create-patient.js (hunk 5 above)
Comment (snehi23) [inline controller/handler/create-patient.js]: ok.

**Comment by ChristianMurphy** [comment]
:shipit:

**Comment by poojaRal** [comment]
LGTM

**Pull request merged**